src/esp/gfx/replay/Player.cpp

```diff
@@ -223,27 +221,78 @@ void Player::applyKeyframe(const Keyframe& keyframe) {
     const auto& instanceKey = pair.first;
     CORRADE_INTERNAL_ASSERT(createdInstances_.count(instanceKey) == 0);
     createdInstances_[instanceKey] = node;
+    creationInfos_[instanceKey] = adjustedCreation;
   }
 
-  if (!frameCleared) {
+  // TODO: Move the code below into renderer-specific implementations.
```

Comment:
@eundersander 

The implementation suggested here is done in the player itself rather than the renderer-specific implementations, therefore goes against the current design.

I did it here because the `RenderAssetInstanceKey` bookkeeping is done here. Otherwise, it would have to be refactored into `AbstractSceneGraphPlayerImplementation` and `BatchPlayerImplementation`.

As previously discussed, we would like to have each renderer to have their own player, I figured that this dynamic cast is an acceptable trade-off.

Let me know if you think that this should be refactored immediately.